Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -634,7 +634,7 @@ public int CompareTo(IDateTime? dt)
     public string ToString(string? format, IFormatProvider? formatProvider)
     {
         formatProvider ??= CultureInfo.InvariantCulture;
-        var dateTimeOffset = DateUtil.ToZonedDateTimeLeniently(Value, _tzId).ToDateTimeOffset();
+        var dateTimeOffset = DateUtil.ToZonedDateTimeLeniently(Value, _tzId ?? string.Empty).ToDateTimeOffset();
```

Comment:
Shouldn't we handle floating separately? Not sure how the method behaves if we pass an empty string.